rfcs/Subworlds.md

```diff
@@ -0,0 +1,139 @@
+  - In order to begin populating level 2's entities, something must exist to distinguish the two sets of entities, so that systems operating for level 1 (e.g. _physics_) do not operate on level 2's entities, even though they may contain physics components.
+- Only having a single `World` requires that systems filter the entities based on what level they belong to, adding overhead and complexity to either the `Query`, the system, the components describing the entities, or a combination.
+- `Subworlds` would provide a clean and efficient solution to this problem. 
+  - The primary systems `[physics, render, etc...]` would only operate on Level 1's `Subworld`.
+  - A `load_next_level` system could be running concurrently on Level 2's `Subworld`. 
+  - Then when Level 2 begins, the primary systems operate the on later `Subworld` while the former one gets saved/dropped.
+
+### Things `Subworlds` would allow 🙂
+- Running systems on subworld `A` on thread `A`, while running the same systems on subworld `B` on thread `B`.
+- Running systems on specific disjoint groups of entities based on some criteria.
+- Running systems on subworld `A` every frame and on subworld `B` every 10 seconds.
+- Helps systems stay clean and not require overuse of _marker-components_ (and possibly _relations_) which can lead to excessive archetypal fracturing which will end up hindering performance. 
+- Give developers the choice to use either a _single_, _global_, world that contains every entity (_note_: for some games, that might makes the most sense and be the most performing), or multiple `Subworlds` if they so desire.
+
+## Reference-level explanation 🧑‍🏫
+
+TODO: expand upon this
+
+```rust
+struct Worlds {
+    subworlds: HashMap<WorldId, World>,
+}
+
+struct World { // aka `Subworld`
+    entities: Entities,
+    components: Components,
+    // more stuff
+}
+```
+
+### Example API (This is not final!) 💻
+```rust
+fn setup(mut commands: Commands) {
+    commands
+        .spawn() // this will spawn the entity into the default world
+        .create_world("World2")
+        .create_world("World3")
+        .set_world("World2") // sets the current world being operated on
+        .spawn(); // will spawn this entity into "World2"
+}
+
+fn on_level_change(mut commands: Commands) {
+    commands
+        .remove_world("World2") // removes the world and all of it's components
+        .add_world("AnotherWorld")
+        .with_system(some_system.system())
+        .unwrap();
+}
+
+fn main() {
+    App::build()
+        // This system will operate on *all* worlds by default
+        .add_system(general_system.system()) 
+        // add this system which only operates on world 2
+        .add_world_system(["World2"], world2_system.system())
+        // add this system which runs on the Default world and World2
+        .add_world_system(["Default", "World2"], two_world_system.system())
```

Comment:
I would love to see a "for each world, run this system" API as well. I expect this will be the common case.

Perhaps with added filtering?